django/db/backends/oracle/schema.py

```diff
@@ -77,11 +76,22 @@ def _alter_field_lob_workaround(self, model, old_field, new_field):
         new_temp_field.column = self._generate_temp_name(new_field.column)
         # Add it
         self.add_field(model, new_temp_field)
+        # Explicit data type conversion
+        # https://docs.oracle.com/cd/B19306_01/server.102/b14200/sql_elements002.htm#sthref340
+        new_value = self.quote_name(old_field.column)
+        if new_field.get_internal_type() == 'TimeField':
+            old_type = old_field.db_type(self.connection)
+            if old_type == 'NCLOB':
+                new_value = "TO_CHAR(%s)" % new_value
+                old_type = 'VARCHAR2'
```

Comment:
According to http://docs.oracle.com/cd/E11882_01/server.112/e41084/functions213.htm#SQLRF06142 this `if` block may be redundant (it mentions specifically `CLOB`, not `NCLOB`, but it would be surprising if they differ)